src/main/java/org/carlmontrobotics/robotcode2023/commands/TeleopDrive.java

```diff
@@ -52,34 +55,45 @@ public void execute() {
     // Used to make sure that the robot does not try to change its angle unless it is moving,
-    if (Math.abs(fwd.getAsDouble()) <= Constants.OI.JOY_THRESH) rawForward = 0.0;
-    else rawForward = maxForward * fwd.getAsDouble();
-    if (Math.abs(str.getAsDouble()) <= Constants.OI.JOY_THRESH) rawStrafe = 0.0;
-    else rawStrafe = maxStrafe * str.getAsDouble();
-
-    rotateClockwise = rcw.getAsDouble();
+    double forward = fwd.getAsDouble();
+    double strafe = str.getAsDouble();
+    double rotateClockwise = rcw.getAsDouble();
+    if (Math.abs(forward) <= Constants.OI.JOY_THRESH) forward = 0.0;
+    else forward *= maxForward;
+    if (Math.abs(strafe) <= Constants.OI.JOY_THRESH) strafe = 0.0;
+    else strafe *= maxStrafe;
     if (Math.abs(rotateClockwise) <= Constants.OI.JOY_THRESH) rotateClockwise = 0.0;
-    else rotateClockwise = maxRCW * rotateClockwise;
-    double targetAccelerationX = (rawForward - currentForward)/deltaT;
-    double targetAccelerationY = (rawStrafe - currentStrafe)/deltaT;
-    double accelerationMagnitude = Math.hypot(targetAccelerationX, targetAccelerationY);
-    if (accelerationMagnitude >= autoMaxAccelMps2) {
-      targetAccelerationX *= autoMaxAccelMps2/accelerationMagnitude;
-      targetAccelerationY *= autoMaxAccelMps2/accelerationMagnitude;
+    else rotateClockwise *= maxRCW;
+
+    // Limit acceleration of the robot
+    double accelerationX = (forward - currentForwardVel) / robotPeriod;
+    double accelerationY = (strafe - currentStrafeVel) / robotPeriod;
+    double translationalAcceleration = Math.hypot(accelerationX, accelerationY);
+    if(translationalAcceleration > autoMaxAccelMps2) {
+      Translation2d limitedAccelerationVector = new Translation2d(autoMaxAccelMps2, Rotation2d.fromRadians(Math.atan2(accelerationY, accelerationX)));
+      Translation2d limitedVelocityVector = limitedAccelerationVector.times(robotPeriod);
+      currentForwardVel += limitedVelocityVector.getX();
+      currentStrafeVel += limitedVelocityVector.getY();
+    } else {
+      currentForwardVel = forward;
+      currentStrafeVel = strafe;
     }
-    currentForward += targetAccelerationX*deltaT;
-    currentStrafe += targetAccelerationY*deltaT;
-    if (Math.abs(currentForward) <= minVelocityMps)
-      currentForward = 0;
-    if (Math.abs(currentStrafe) <= minVelocityMps)
-      currentStrafe = 0;
-    double driveMultiplier = slow.getAsBoolean() ? kSlowDriveSpeed : 1;
-    double rotationMultiplier = slow.getAsBoolean() ? kSlowDriveRotation : 0.55;
-    return new double[] {currentForward * driveMultiplier, currentStrafe * driveMultiplier, -rotateClockwise * rotationMultiplier};
+
+    // ATM, there is no rotational acceleration limit
+
+    // If the above math works, no velocity should be greater than the max velocity, so we don't need to limit it.
+
+    // Quick-Stop
+    if (Math.abs(forward) <= minVelocityMps) currentForwardVel = 0;
+    if (Math.abs(strafe) <= minVelocityMps) currentStrafeVel = 0;
```

Comment:
I recommend against doing this because it could cause tipping when decelerating.

Reply:
This was added as a driver request. I figured I would port it over to this version. I don't think they thought it was that important tho, so I'll remove it.